### PR TITLE
support --noImplicitAny

### PR DESCRIPTION
**Pull request by timjacobi**
This code adds support for the `noImplicitAny` compiler option. It looks for a TypeScript config file and parses it. If `noImplicitAny` is set it adds a typecast to the generated output.

### CODE REVIEW

File: src/loadCompilerOptions.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  }
+
+  const result = ts.readConfigFile(configFileName, ts.sys.readFile);
+  if (result.error) {
+    let message;
+    if (typeof result.error.messageText === "string") {
+      message = result.error.messageText;
+    } else {
+      message = result.error.messageText.messageText;
+    }
```
Comment (alloy) [inline src/loadCompilerOptions.ts]: What is this doing and could you leave a comment about it?

File: src/loadCompilerOptions.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import * as ts from "typescript";
+
+export const loadCompilerOptions = (): ts.CompilerOptions => {
+  const configFileName = ts.findConfigFile(process.cwd(), ts.sys.fileExists);
+  if (!configFileName) {
+    throw new Error("Could not find config file!");
+  }
```
Comment (alloy) [inline src/loadCompilerOptions.ts]: I think that in this case we should just default to disabling features that are dependent on knowing compiler options.

**Comment by timjacobi** [comment]
I didn't change any dependencies. Which changes would you like me to separate?

File: src/loadCompilerOptions.ts (hunk 1 above)
Comment (timjacobi) [reply src/loadCompilerOptions.ts]: According to TypeScript the type of `result.error.messageText` is `string | DiagnosticMessageChain` and `DiagnosticMessageChain` in turn has a `messageText` property. I was wondering if we should just return an empty object from this function if we can't find or read the config file. What do you think?

Comment (alloy) [reply src/loadCompilerOptions.ts]: Yeah I agree 👍, I feel like any real issues with the config file (such as syntax errors) will most likely be surfaced in a better way while using any other TS tooling anyways.

**Comment by alloy** [comment]
> I didn't change any dependencies. Which changes would you like me to separate?

I'm referring to all the `yarn.lock` changes. Did you not run `yarn upgrade` or something?

**Comment by timjacobi** [comment]
I don't think I ran `yarn upgrade`. I'll just revert the change.

**Comment by alloy** [comment]
Perfect, thanks for the great contribution!

**Pull request merged**